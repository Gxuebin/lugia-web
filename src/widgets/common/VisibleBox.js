/*
 * 可通过visible属性控制是否可见的盒子
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';

type VisibleBoxProps = {
  visible: boolean
}
export default styled.div`
   ${(props: VisibleBoxProps) => (props.visible ? '' : 'display: none;')}
`;
