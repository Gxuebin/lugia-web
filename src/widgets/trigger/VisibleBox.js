/*
 * 可通过visible属性控制是否可见的盒子
 */
import React from 'react';
import styled from 'styled-components';

export default styled.div`
  display: ${props => (props.visible ? '' : 'none')};
`;
