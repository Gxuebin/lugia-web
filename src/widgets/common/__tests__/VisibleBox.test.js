//@flow
import React from 'react';
import chai from 'chai';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import VisibleBox from '../VisibleBox';

const { expect: exp } = chai;

describe('VisibleBox', () => {

  it('visible  is true', () => {
    const visibleBox = renderer.create(<VisibleBox visible></VisibleBox>).toJSON();
    expect(visibleBox).toMatchSnapshot();

  });

  it('visible is false', () => {
    const visibleBox = renderer.create(<VisibleBox visible={false}></VisibleBox>).toJSON();
    expect(visibleBox).toMatchSnapshot();
    expect(visibleBox).toHaveStyleRule('display', 'none');
  });
  it('default visible default is false', () => {
    const visibleBox = renderer.create(<VisibleBox></VisibleBox>).toJSON();
    expect(visibleBox).toMatchSnapshot();
    expect(visibleBox).toHaveStyleRule('display', 'none');
  });

});
