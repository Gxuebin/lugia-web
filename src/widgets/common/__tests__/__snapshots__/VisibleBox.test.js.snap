// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibleBox default visible default is false 1`] = `
.c0 {
  display: none;
}

<div
  className="c0"
/>
`;

exports[`VisibleBox visible  is true 1`] = `
<div
  className=""
/>
`;

exports[`VisibleBox visible is false 1`] = `
.c0 {
  display: none;
}

<div
  className="c0"
/>
`;
